app/models/customer.rb

```diff
@@ -52,10 +52,12 @@ def associate_user
     self.user = user || Spree::User.find_by(email: email)
   end
 
-  def check_for_orders
-    return true unless orders.any?
-
-    errors.add(:base, I18n.t('admin.customers.destroy.has_associated_orders'))
```

Comment:
That translation key can also be removed (when removing the reference in the spec). I would have done that in the same commit.